learnosity_sdk/request/dataapi.py

```diff
@@ -60,8 +60,12 @@ def results_iter(self, endpoint, security_packet,
         for response in self.request_iter(endpoint, security_packet,
                                           secret, request_packet,
                                           action):
-            for result in response['data']:
-                yield result
+            if isinstance(response['data'], dict):
+                for key, value in Future.iteritems(response['data']):
```

Comment:
No need for `Future`:

```suggestion
                for key in response['data']:
                  yield {key: response['data'][key]}
```